x-pack/plugins/enterprise_search/public/applications/app_search/components/relevance_tuning/relevance_tuning_logic.ts

```diff
@@ -497,7 +504,11 @@ export const RelevanceTuningLogic = kea<
       const { searchSettings } = values;
       const { boosts } = searchSettings;
       const updatedBoosts = cloneDeep(boosts[name]);
-      updatedBoosts[boostIndex][optionType] = value;
+      if (optionType === 'operation') {
+        updatedBoosts[boostIndex][optionType] = value as BoostOperation;
+      } else {
+        updatedBoosts[boostIndex][optionType] = value as BoostFunction;
+      }
```

Comment:
Just curious, I legit have no idea - does `updatedBoosts[boostIndex][optionType] = value as BoostOperation | BoostFunction` work or nah? No worries if not, maybe @byronhulcher might also have some ideas?

Reply:
No dice with that. Which is weird, because I can do something similar in TS Playground. Not the first time where I could do something in TS Playground but not here: https://www.typescriptlang.org/play?ssl=1&ssc=1&pln=17&pc=71#code/KYDwDg9gTgLgBMAdgVwLZwGLMQYxgSwkQEMAbAIQggGcYtcCi4BvAWACg44BxY5a6vmKI4AXjgByAOZ8BQxBIA0HLgFFwRJATJjJoSIi1DSSlXAAy+Q8Si6JpK8Bun2AXw4d90eEjRwAClAQIPio+DAAnpQ0dNh4hCJsnDyygsJ2pBAyUOEAFmE4LmoahojapHZepeVFFo42GfVQLu7sHJFgwHDRtPTxTOJ9jCQUVL1xw3AAPgFBIWGRPbEMCQDcHuykwPDEAFyYEwlkS0MJ07PBoeFRY8v9iOtt7MQZWTZ5BRJwxNQHK0THW6nJgzQKXBY3GLAh5AA

Reply:
Gotcha, no worries. It doesn't _excessively_ bother me to be adding branching logic for the sake of typing but it does feel a bit weird. Leaving this uncollapsed in case anyone has any other ideas or if we come back to this in the future at some point, but no change requested

Reply:
Definitely, ty.